Makefile

```diff
@@ -126,32 +126,40 @@ SED ?= $(shell which gsed 2>/dev/null || which sed)
 
 %/$(UPTODATE_RACE): GOOS=linux
 %/$(UPTODATE_RACE): %/Dockerfile
-	# Build Dockerfile.distroless if it exists, we use distroless/base-nossl-debian12 as base image since race detector needs glibc packages.
-	if [ -f $(@D)/Dockerfile.distroless ]; then \
-		$(SUDO) docker build -f $(@D)/Dockerfile.distroless \
+	# Build Dockerfile.alpine if it exists
+	if [ -f $(@D)/Dockerfile.alpine ]; then \
+		$(SUDO) docker build -f $(@D)/Dockerfile.alpine \
 			--build-arg=revision=$(GIT_REVISION) \
 			--build-arg=goproxyValue=$(GOPROXY_VALUE) \
 			--build-arg=USE_BINARY_SUFFIX=true \
 			--build-arg=BINARY_SUFFIX=_race \
-			--build-arg=BASEIMG="gcr.io/distroless/base-nossl-debian12" \
-			-t $(IMAGE_PREFIX)$(shell basename $(@D))-distroless:$(IMAGE_TAG_RACE) $(@D)/; \
+			--build-arg=EXTRA_PACKAGES="gcompat" \
+			-t $(IMAGE_PREFIX)$(shell basename $(@D))-alpine:$(IMAGE_TAG_RACE) $(@D)/; \
 	fi;
 	@echo
 	# We need gcompat -- compatibility layer with glibc, as race-detector currently requires glibc, but Alpine uses musl libc instead.
-	$(SUDO) docker build --build-arg=revision=$(GIT_REVISION) --build-arg=goproxyValue=$(GOPROXY_VALUE) --build-arg=USE_BINARY_SUFFIX=true --build-arg=BINARY_SUFFIX=_race --build-arg=EXTRA_PACKAGES="gcompat" -t $(IMAGE_PREFIX)$(shell basename $(@D)):$(IMAGE_TAG_RACE) $(@D)/
+	# for grafana/mimir which is based on distroless image, the EXTRA_PACKAGES is just ignored.
```

Comment:
Why do we even pass `EXTRA_PACKAGES` to the build then? We can remove it.

Reply:
since this is used in other dockerfile still, not only for cmd/mimir, we still need it.

Reply:
I see what you mean now,  it is UPTODATE_RACE target which is only for cmd/mimir, removing it.